File: Problem Solving/programmers/76502.js
function isCorrect(s) {
    const open = new Array();
    const sz = s.length;

    for (let i = 0; i < sz; i++) {
        if (s[i] === '[' || s[i] === '(' || s[i] === '{') {
            open.unshift(s[i]);
        }
        else {
            if (open.length === 0) return false;
            const c = open[0];
            open.shift();

            if ((c !== '[' && s[i] === ']') ||
                (c !== '(' && s[i] === ')') ||
                (c !== '{' && s[i] === '}')) return false;
        }
    }

    if (open.length !== 0) return false;
    return true;
}

function solution(s) {
    const sz = s.length;
    let answer = 0;

    for (let i = 0; i < sz; i++) {
        if (isCorrect(s)) answer++;
        s = s.concat(s[0]).slice(1);
    }

    return answer;
}

File: Problem Solving/programmers/76502.cpp
#include <iostream>
#include <string>
#include <vector>
#include <stack>
#include <queue>
using namespace std;

bool isRight(queue<char> s) {
    stack<char> open;
    int sz = s.size();
    
    for(int i=0;i<sz;i++){
        char c = s.front();
        s.pop();
        
        if(c == '[' || c == '(' || c == '{'){
            open.push(c);
        }
        else {
            if(open.empty()) return false;
            char t = open.top();
            open.pop();
            
            if((t != '[' && c == ']') ||
               (t != '(' && c == ')') ||
               (t != '{' && c == '}')) return false;
        }
    }
    
    if(!open.empty()) return false;
    return true;
}

int solution(string s) {
    int answer = 0;
    queue<char> strQueue;
    
    for(int i=0;i<s.length();i++){
        strQueue.push(s[i]);
    }
    
    for(int i=0;i<s.length();i++){
        if(isRight(strQueue)) answer++;
        
        char c = strQueue.front();
        strQueue.pop();
        strQueue.push(c);
    }
    
    return answer;
}